packages/flame/lib/src/components/mixins/has_paint.dart

```diff
@@ -110,9 +112,28 @@ mixin HasPaint<T extends Object> on Component implements OpacityProvider {
     }
   }
 
+  /// Creates an [OpacityProvider] for given [paintId] and can be used as
+  /// `target` for [OpacityEffect].
   OpacityProvider opacityProviderOf(T paintId) {
     return _ProxyOpacityProvider(paintId, this);
   }
+
+  /// Creates an [OpacityProvider] for given list of [paintIds] and can be
+  /// used as `target` for [OpacityEffect].
+  ///
+  /// When opacities of all the given [paintIds] are not same, this provider
+  /// directly effects opacity of the most opaque paint. Additionally, it
+  /// modifies other paints such that their respective opacity ratio with most
+  /// opaque paint is maintained.
+  ///
+  /// If [paintIds] is null or empty, all the paints are used for creating the
+  /// [OpacityProvider].
+  ///
+  /// Note: Each call results in a new [OpacityProvider] and hence the cached
+  /// opacity ratios are calculated using opacities when this method was called.
+  OpacityProvider opacityProviderOfList({List<T>? paintIds}) {
+    return _MultiPaintOpacityProvider(paintIds, this);
```

Comment:
well `paintIds` is null, this should default to using all paints, as the description implies.